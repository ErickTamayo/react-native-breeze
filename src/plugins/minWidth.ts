import { PluginFunction, PluginPattern } from "./types";

export type PluginGroups = {
  key: string;
};

export const pattern: PluginPattern = ({ keys }) => {
  return new RegExp(`^min-w-(?<key>${keys("minWidth")})$`);
};

export const plugin: PluginFunction<PluginGroups> = ({ groups, theme }) => {
  const { key } = groups;
  const value = theme(["minWidth", key]);
  return { minWidth: value };
};
