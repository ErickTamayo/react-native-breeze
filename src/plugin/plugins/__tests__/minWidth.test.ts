import {
  generateInput,
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../utils/tests";
import { pattern, plugin } from "../minWidth";

describe("minWidth", () => {
  // prettier-ignore
  const shouldMatch = generateInput('min-w', ['0', 'full']);

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["min-w-unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
