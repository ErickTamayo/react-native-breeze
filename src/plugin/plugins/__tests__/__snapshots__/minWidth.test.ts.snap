// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`minWidth Should parse (min-w-0) correctly 1`] = `
Object {
  "minWidth": 0,
}
`;

exports[`minWidth Should parse (min-w-full) correctly 1`] = `
Object {
  "minWidth": "100%",
}
`;
